File: app/src/main/java/com/darkrockstudios/apps/adventurephotoframe/api/PhotoFrameApi.java
package com.darkrockstudios.apps.adventurephotoframe.api;

import com.darkrockstudios.apps.adventurephotoframe.data.Photo;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Path;
import retrofit2.http.Query;

/**
 * Created by adamw on 12/21/2016.
 */

public interface PhotoFrameApi
{
	@GET("photoframe/{photoFrameId}")
	Call<Photo> getPhoto( @Path("photoFrameId") long photoFrameId );

	@POST("health/{photoFrameId}")
	Call<Void> healthCheckIn( @Path("photoFrameId") long photoFrameId, @Query("version") int version );
}

File: app/src/main/java/com/darkrockstudios/iot/adventurephotoframe/api/PhotoFrameApi.kt
package com.darkrockstudios.iot.adventurephotoframe.api

import com.darkrockstudios.iot.adventurephotoframe.data.Photo
import com.darkrockstudios.iot.adventurephotoframe.data.RegistrationResponse
import retrofit2.Call
import retrofit2.http.*

/**
 * Created by adamw on 12/21/2016.
 */

interface PhotoFrameApi
{
	@GET("photoframe/{deviceId}")
	fun getPhoto(@Path("deviceId") deviceId: String): Call<Photo>

	@GET("photoframe/{deviceId}/{photoId}")
	fun getPhotoById(@Path("deviceId") deviceId: String, @Path("photoId") photoId: Long): Call<Photo>

	@FormUrlEncoded
	@POST("health/{photoFrameId}")
	fun healthCheckIn(@Path("photoFrameId") photoFrameId: Long,
					  @Query("deviceId") deviceId: String,
					  @Query("version") version: Int,
					  @Field("update_channel") updateChannel: String): Call<Void>

	@FormUrlEncoded
	@POST("photowallnfc/{deviceId}")
	fun photoWallNfc(@Path("deviceId") deviceId: String, @Field("registration_id") registrationId: String): Call<RegistrationResponse>
}
